engines/osxnative/sandboxbuilder.go

```diff
@@ -0,0 +1,33 @@
+package osxnative
+
+import (
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+type sandboxbuilder struct {
+	engines.SandboxBuilderBase
+	env         map[string]string
+	taskPayload *payload
+	context     *runtime.TaskContext
+}
+
+func (s sandboxbuilder) SetEnvironmentVariable(name string, value string) error {
+	_, ok := s.env[name]
```

Comment:
I assume we're doing this because if it's already set, then there is an issue?

nit: could "ok" be named something better, like "exists" which is what we're checking here?